### PR TITLE
ONE: Added card "Awaken the Sleeper" and its abilities and effects

### PR DESCRIPTION
**Pull request by Merlingilb**

### CODE REVIEW

**Comment by miesma** [comment]
Doesn't that also "Destroy" indestructible equipment as you just move all equipment to the GY? There exists a 

DestroyAllAttachedToTargetEffect which call .destroy()

Ref. Blasfire Bolt

**Comment by Merlingilb** [comment]
> Doesn't that also "Destroy" indestructible equipment as you just move all equipment to the GY?

Yes I did not think about that. Thank you.
I've adjusted the effect now to youse the DestroyAllAttachedToTargetEffect

File: Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public boolean apply(Game game, Ability source) {
+        Permanent targetCreature = game.getPermanent(source.getFirstTarget());
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null || targetCreature == null) {
+            return false;
+        }
+        if (targetCreature.getAttachments().size() > 0) {
+            if (player.chooseUse(Outcome.DestroyPermanent, "Destroy all equipped enchantments?", source, game)) {
+                DestroyAllAttachedToTargetEffect destroyAllAttachedToTargetEffect =
+                        new DestroyAllAttachedToTargetEffect(StaticFilters.FILTER_PERMANENT_EQUIPMENT,
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java:74]: Destroying Attached instead Equipment

File: Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class TargetPermanentIsEquippedCondition implements Condition {
+    public TargetPermanentIsEquippedCondition() {
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (!source.getTargets().isEmpty()) {
+            Permanent permanent = game.getPermanent(source.getFirstTarget());
+            if (permanent != null) {
+                return permanent.getAttachments().size() > 0;
```
Comment (JayDi85) [inline Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java:17]: Checking Attachments instead Equipped

**Review by JayDi85 — COMMENTED** [review]
Code example from other place:

shot_230302_002657

File: Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java (hunk 1 above)
Comment (Merlingilb) [reply Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java:74]: But with the filter it should only be equipments right?

File: Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java (hunk 2 above)
Comment (miesma) [reply Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java:17]: Right would destroy enchantments aswell

Comment (Merlingilb) [reply Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java]: i changed this now to only check for equipments

File: Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java
Comment (theelk801) [inline Mage/src/main/java/mage/abilities/condition/common/TargetPermanentIsEquippedCondition.java:1]: you don't need this condition, there's already a custom effect that handles it

File: Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{R}");
+
+        //Gain control of target creature until end of turn. Untap that creature. It gains haste until end of turn.
+        //If it's equipped, you may destroy all Equipment attached to that creature.
+        this.getSpellAbility().addEffect(new GainControlTargetEffect(Duration.EndOfTurn));
+        this.getSpellAbility().addEffect(new UntapTargetEffect().setText("Untap that creature"));
+        this.getSpellAbility().addEffect(new GainAbilityTargetEffect(
+                HasteAbility.getInstance(), Duration.EndOfTurn
+        ).setText("It gains haste until end of turn."));
+        this.getSpellAbility().addEffect(new ConditionalOneShotEffect(
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AwakenTheSleeper.java:37]: this doesn't need to be a `ConditionalOneShotEffect`, you're already using a custom effect

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**